mock/mock.go

```diff
@@ -335,6 +332,7 @@ func (m *Mock) MethodCalled(methodName string, arguments ...interface{}) Argumen
 	if call.WaitFor != nil {
 		<-call.WaitFor
 	}
+	time.Sleep(call.WaitTime)
```

Comment:
Rather than doing `time.Sleep`, do `call.WaitFor = time.After(call.waitTime)` before the if.

Otherwise it might be possible to set two different waiting times for the same call.

Reply:
Then in that case we would be losing the channel which is set via `time.WaitUntil` (which i can control when to send data not just with timer.

Reply:
That is not allowed right now anyway, since `After` currently calls `WaitUntil` and overrides the property.

The change I mention would be aligned with how things work right now.

By the way, remember to lock/unlock the mutex in `After`.

Reply:
@ernesto-jimenez i've added `time.Sleep(waitTime)` if `waitFor` is nil, adding it before would make the duration override, in this case channel is used. Anyway It depends on the client usage `WaitUntil(ch).After(d) or .After(d).WaitUntil(ch)` and not defined in here. so i hope this is fine.